### PR TITLE
Extensions

### PR DESCRIPTION
**Pull request by elprans**
Add initial (provisional) support for `CREATE EXTENSION PACKAGE` and `CREATE EXTENSION` and add formal declarations for GraphQL et al.

### CODE REVIEW

File: edb/lib/ext/graphql.edgeql [hunk 1]
```diff
@@ -17,6 +17,8 @@
 #
 
 
+CREATE EXTENSION PACKAGE graphql VERSION '1.0';
+
 CREATE MODULE stdgraphql;
```
Comment (1st1) [inline edb/lib/ext/graphql.edgeql:22]: Should this line be nested in `CREATE EXTENSION PACKAGE`?

File: edb/schema/schema.py [hunk 2]
```diff
@@ -52,16 +52,25 @@
...
     sn.UnqualName('schema'),
     sn.UnqualName('math'),
     sn.UnqualName('sys'),
     sn.UnqualName('cfg'),
     sn.UnqualName('cal'),
+    sn.UnqualName('stdgraphql'),
 )
-STD_MODULES = frozenset(STD_LIB + (sn.UnqualName('stdgraphql'),))
 
+STD_SOURCES = (
```
Comment (1st1) [inline edb/schema/schema.py:65]: Can you please add a comment what this is used for?

File: edb/server/dbview/dbview.pyx [hunk 3]
```diff
@@ -555,7 +555,7 @@ cdef class DatabaseIndex:
         self._comp_sys_config = config.get_compilation_config(sys_config)
 
     def get_db(self, dbname):
-        return self._dbs[dbname]
+        return self._dbs.get(dbname)
```
Comment (1st1) [inline edb/server/dbview/dbview.pyx:558]: I used `[]` on purpose here, to let it fail (it never should fail, unless there's a bug).

**Review by 1st1 — APPROVED** [review]

File: edb/server/dbview/dbview.pyx (hunk 3 above)
Comment (elprans) [reply edb/server/dbview/dbview.pyx:558]: `POST /db/nonexistent/edgeql` crashes with a `KeyError` currently, and I don't think that catching `KeyError` is nicer than checking for `None`.  We can add another method: `maybe_get_db` or something.

File: edb/lib/ext/graphql.edgeql (hunk 1 above)
Comment (elprans) [reply edb/lib/ext/graphql.edgeql:22]: I haven't actually implemented nested DDL for extensions, as we need to add a new tracking mechanism to attribute created objects to extensions for it to make sense, and that, in turn, requires a bit of refactoring.  Can be done post beta.

File: edb/server/dbview/dbview.pyx (hunk 3 above)
Comment (1st1) [reply edb/server/dbview/dbview.pyx:558]: Well, you might as well then fix connecting to a non-existent DB via binary.

Comment (1st1) [reply edb/server/dbview/dbview.pyx:558]: Returning None here might make it a segfault now. Please check all get_db() callers.

Comment (1st1) [reply edb/server/dbview/dbview.pyx:558]: Also, I wouldn't return None. I'd raise a proper `edb.errors` exception here, or at least a LookupError.

Comment (elprans) [reply edb/server/dbview/dbview.pyx:558]: Done.

**Pull request merged**